overlord/ifacestate/handlers.go

```diff
@@ -452,14 +453,16 @@ func (m *InterfaceManager) doDisconnect(task *state.Task, _ *tomb.Tomb) error {
 	if err != nil {
 		return err
 	}
+	cref := &interfaces.ConnRef{PlugRef: plugRef, SlotRef: slotRef}
 
+	remapIncomingConnRef(st, cref)
```

Comment:
we don't have a corresponding call in doConnect (I see we do calls anyway in Connect/Disconnect themselves because of conflicts, doubling them is ok I think, but not being symmetric is confusing)

Afaict  the undo bit are missing remappings no?

Reply:
also *DiscardConns needs changes?   if we are discarding snapd, we should look for "core" conns, no?

Reply:
The undo bits are indeed missing remappings. I think this branch should wait a little bit and perhaps only go in with the bigger patch that enables snapd to hold interfaces because then we can do proper round of unit tests that check each of the cases. Right now much of the new code is untested in longer realistic scenarios.